layouts/partials/contributors.html

```diff
@@ -0,0 +1,146 @@
+  <div id="contributors-slide">
+    <div class="contributors-row"></div>
+    <div class="contributors-row" style="position:absolute;"></div>
+    <div class="contributors-row" style="position:absolute;"></div>
+    <div class="contributors-row" style="position:absolute;"></div>
+    <div class="contributors-row" style="position:absolute;"></div>
+  </div>
+</section>
+
+{{/*
+  Pull in the contributors array into a hugo var at build time.
+  It's rendered inline as a JS array. We build out the content after the page
+  has rendered to avoid loading 400+ images on intial load.
+  The data looks like:
+  ```js
+  [
+    {"url":"https://github.com/0intro","photo":"https://avatars.githubusercontent.com/u/6043744?v=3","username":"0intro"}
+  ]
+  ```
+*/}}
+{{ $contributors := getJSON "https://contributors.cloud.ipfs.team/contributors?org=all" }}
+<script>
+  (function () {
+    var contributors = {{ $contributors }}
+    var hexWidth = 62
+    var hexHeight = 72
+    var loadingDelayMs = 150
+    var containerEl = document.getElementById('contributors-slide')
+    var containerElTop = containerEl.getBoundingClientRect().top
+    var screenHeight = window.innerHeight
+    var pageYOffset = window.pageYOffset
+    var inView = (pageYOffset + screenHeight) > containerElTop
+    if (inView) {
+      renderContributors(contributors)
+    } else {
+      window.addEventListener('scroll', isContributorsSectionInView)
+    }
+
+    function isContributorsSectionInView () {
+      inView = (window.pageYOffset + screenHeight) > containerElTop
+      if (inView) {
+        renderContributors(contributors)
+        window.removeEventListener('scroll', isContributorsSectionInView)
+      }
+    }
+
+    function renderContributors (data) {
+      containerEl.className = 'running'
+      var rows = document.querySelectorAll('.contributors-row')
+      var rowWidth = Math.ceil(data.length / rows.length) * hexWidth
+      ;[].forEach.call(rows, function (row, i) {
+        row.style.height = hexHeight + 'px'
+        row.style.width = rowWidth + 'px'
+        row.style.top = i * hexHeight * 0.75 + 'px'
+        if ((i % 2) === 1) {
+          row.style.left = hexWidth / 2 + 'px'
+        }
+      })
+      var nextRow = 0
+      function renderWithPause(itemIndex, lastRender) {
+        renderContributor(data[itemIndex], function (err, el) {
+          rows[nextRow].appendChild(el)
+          setTimeout(function () {
+            el.className = 'hex show'
+          }, 10)
+          nextIndex = itemIndex + 1
+          if (nextIndex === data.length) return // All done.
+          nextRow = (nextRow + 1) % rows.length
+          var now = Date.now()
+          var diff = now - lastRender
+          var delay = loadingDelayMs - diff
+          if (delay < 0) delay = 0
+          setTimeout(function () {
+            renderWithPause(itemIndex + 1, now)
+          }, delay)
+        })
+      }
+      renderWithPause(0, Date.now())
```

Comment:
Feels like you could extract this control flow logic from the rendering and publish as a module ;)